src/components/TodoContext.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+  Dispatch,
+  SetStateAction,
+  createContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+
+const todosTemp: Todo[] = [
+  { id: 23, title: '123', completed: false },
+  { id: 24, title: '567', completed: true },
+];
+
+interface TodoContextType {
+  todos: Todo[];
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+}
+
+export const TodoContext = createContext<TodoContextType>({
+  todos: todosTemp,
+  setTodos: () => {},
+});
+
+type TodoProviderProps = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<TodoProviderProps> = ({ children }) => {
```

Comment:
```suggestion
export const TodoProvider: React.FC<PropsWithChildren> = ({ children }) => {
```
You can import it (PropsWithChildren) from 'react' - FYI